src/parser/mod.rs

```diff
@@ -633,6 +633,14 @@ parser!(match_branch _loc -> 'b (Ast<'b>, Ast<'b>) =
     (pattern, branch)
 );
 
+parser!(without_else_expr _loc =
+    _ <- maybe_newline;
+    expr <- block_or_statement;
+    _ <- expect(Token::Else);
+    otherwise !<- block_or_statement;
+    otherwise
+);
```

Comment:
The `if_expr` parser above already parses a `if <block_or_statement> then <block_or_statement>` before this so using this parser afterward would parse:
```rs
if <block_or_statement>
then <block_or_statement>
<block_or_statement> else <block_or_statement>
```
We don't need to change the `if_expr` parser for this parser, it should continue to parse `if a then b else c`, the `else` expression added in this PR can be a separate binary operator expression entirely.